cyhostboot/src/cyhostboot.c

```diff
@@ -189,8 +247,24 @@ int main(int argc, char **argv)
 	if (action == PROGRAM)
 		printf("Programing file %s\n", args_info.file_arg);
 
+	if (args_info.key_given )
+	{
+		for( int i=0; i<12; i+=2 )
+		{
+			unsigned char k = 0;
+			unsigned char *c = (unsigned char *)&args_info.key_arg[i];
+
+			k|=hex_decode( c[0] )<<4;
+			k|=hex_decode( c[1] );
```

Comment:
You should use strtol with a base 16. It will simplify this parsing.

Reply:
done.